### DISCUSSION TITLE
New SK6812 with 3.3V MCU support

### DISCUSSION THREAD

**Discussion by protyposis** [awawa-dev/HyperHDR#726] [General]
Just wanted to let you all know that the new version of BTF's SK6812 5V LED strip, which is shipped since mid of December 2023, works great without a level shifter - I use it directly with a ESP32 D1 Mini and a 1.5 meters long data line between the ESP and the LED strip. So the HyperHDR setup just got a little simpler (and cheaper) :)

The new ones can be easily identified by the 3rd metal pin (circled red), when looking at the strip from aside:
![image](https://github.com/awawa-dev/HyperHDR/assets/189372/7dfd623a-2c5e-4c32-ae8e-fbda6cc6510d)

Tested and confirmed working with
- https://github.com/awawa-dev/HyperSerialESP32 + ESP32 D1 Mini
- https://github.com/awawa-dev/HyperSerialPico + Raspberry Pi Pico

**Comment by awawa-dev** [comment]
Not so fast 😃 I noticed it earlier and commented https://github.com/awawa-dev/HyperHDR/discussions/561#discussioncomment-8011515 [_full tech specs: trust me bro, or this short clip on YouTube_] 
When will I have it, test it and we'll see. I remember the story with HD108 which, depending on the author of the next iteration of the documentation, either had support for MCU3.3V or did not have it (and in addition, it did not have a video on YouTube 😉  )

**Reply by protyposis** [reply]
Whether you believe it or not, I am holding the LED strip right in my hands, and it clearly works with 3.3V. Ordered from BTF on Aliexpress on December 21.

**Reply by awawa-dev** [reply]
The problem is that ordinary 'old' sk6812 also could work with 3.3V: well... sometimes. The problem is tolerance/range (there is no word about it for 'new' sk6812). Because if the data line is a bit longer, the 'old' sk6812 suddenly stopped to work. I need to see 'new' tech specification that supports their claims. Personally I have rather a bad experience with BTF led strips.

**Reply by protyposis** [reply]
I understand, you want written proof. I just thought I'll report my experience here, since I haven't seen any reports on the new version yet. Even without tech specs, it might be of help to someone.

I also have the old version from BTF. The old version doesn't work correctly (erratic behavior, random colors) with the very same ESP32 board (without level shifter) that flawlessly drives the new version.

**Reply by awawa-dev** [reply]
If they actually guaranteed at least 2.7V as the lower limit, it would be perfect, even with 3V it would probably be fine if they actually stuck to it. Well, we'll see... not only the recent premiere of these LEDs was ahead of the documentation, because, for example, RPi5 is still waiting for its full version.

**Reply by d22b** [reply]
Thanks for sharing. I am actually looking for an SK6812 led strip with 12V support in Cold White. Any recommendation where to buy?

Thanks!

**Reply by EspressoTini** [reply]
> I understand, you want written proof. I just thought I'll report my experience here, since I haven't seen any reports on the new version yet. Even without tech specs, it might be of help to someone.
> 
> I also have the old version from BTF. The old version doesn't work correctly (erratic behavior, random colors) with the very same ESP32 board (without level shifter) that flawlessly drives the new version.

you don't need level shifting with esp32 since it's 5v tolerant, just add a pullup to 5v.  the same goes for most 3v3 arm chips STM32 etc.

WS2811 use 1k pullups to 5v on the data line, no need for anything else.
UART use a 10k pullup to 5v on the esp32 TX, for the RX it will just accept 5v RX from the other device.
I2C use 4.7 pullups to 5v on both lines as normal.

why the internet does not know this is a mystery to me.  **it is why 5v tolerant pins on 3v3 mcu exists**.

**Reply by awawa-dev** [reply]
One rabbi will say yes, the other will say no... in the case of esp32, we adopt a safe position such eg. as WLED and require a level shifter. We do not assume that there are any safe pins that are 5V tolerant because, at least so far, we have not been able to confirm this in the official documentation from espressif in the case of classic ESP32. I recommend caution so as not to end up like in the case of RPi5, where the message was that some Rpi5 GPIO pins are 5v tolarable. Unfortunately, there was a small catch that very detailed conditions had to be met first. As it turned out, it was so difficult to meet or people simply didn't understand it fully that they ended up with an RPi5 with a burnt GPIO.

**Reply by EspressoTini** [reply]
if it's not officially supported i have to agree with you for production designs that it's not a good idea.

now for the hobbyists, without any reports of damage to my knowledge, and as long as the hobbyist(s) are warned, i think the juice is worth the squeeze.  it's so much easier to wire a pullup, and the esp boards are so affordable.

**Comment by protyposis** [comment]
Also works with the Raspberry Pi Pico

**Comment by awawa-dev** [comment]
For proper testing a level shifter should be used for example. But not to uplevel voltage to 5v but to lower data signal to ~2.7v

**Comment by protyposis** [comment]
I cannot pinpoint the exact reason, but after replacing my Raspberry Pi 4 2GB with the 4GB version and powering the Pi4 and Pico through a separate adapter instead of the LED PSU, the LEDs no longer run stable. Powering the Pico with the LED PSU also didn't help. Replacing the Pi Pico with an Adafruit ItsyBitsy RP2040 with 5V level shifter made them run stable again.

So it seems I just got lucky with my original setup without a level shifter (LEDs, Pi4, Pico powered by single MeanWell PSU at ~5.2V). If you don't want to waste multiple hours with wiring experiments, just use a level shifter!
